File: mycollab-mobile/src/main/scala/com/mycollab/mobile/module/project/view/parameters/ProjectScreenData.scala
package com.mycollab.mobile.module.project.view.parameters

import com.mycollab.module.project.domain.SimpleProject
import com.mycollab.vaadin.mvp.ScreenData
import com.mycollab.common.domain.criteria.ActivityStreamSearchCriteria

/**
  * @author MyCollab Ltd
  * @since 5.0.9
  */
object ProjectScreenData {

  class Goto(param: Integer) extends ScreenData[Integer](param) {}

  class Add(param: SimpleProject) extends ScreenData[SimpleProject](param) {}

  class Edit(param: Integer) extends ScreenData[Integer](param) {}

  class ProjectActivities(param: ActivityStreamSearchCriteria) extends ScreenData[ActivityStreamSearchCriteria](param) {}

  class GotoDashboard() extends ScreenData[AnyRef](null) {}

  class AllActivities(param: ActivityStreamSearchCriteria) extends ScreenData[ActivityStreamSearchCriteria](param) {}

}

File: mycollab-mobile/src/main/java/com/mycollab/mobile/module/project/view/parameters/ProjectScreenData.kt
package com.mycollab.mobile.module.project.view.parameters

import com.mycollab.common.domain.criteria.ActivityStreamSearchCriteria
import com.mycollab.module.project.domain.SimpleProject
import com.mycollab.vaadin.mvp.ScreenData

/**
 * @author MyCollab Ltd
 * @since 6.0.0
 */
class ProjectScreenData {
    class Goto(param: Int) : ScreenData<Int>(param)

    class Add(param: SimpleProject) : ScreenData<SimpleProject>(param)

    class Edit(param: Int) : ScreenData<Int>(param)

    class ProjectActivities(param: ActivityStreamSearchCriteria) : ScreenData<ActivityStreamSearchCriteria>(param)

    class GotoDashboard() : ScreenData<*>(null)

    class AllActivities(param: ActivityStreamSearchCriteria) : ScreenData<ActivityStreamSearchCriteria>(param)
}